client/src/hooks/useDetectDevice.ts

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from 'react'
+
+export const useDetectDevice = () => {
+  const device = {
+    mobile: 'mobile',
+    tablet: 'tablet',
+    desktop: 'desktop',
+  }
```

Comment:
Consider making this an enum exported from this module. Perhaps something like..

```ts
export enum Device {
  Mobile = 'mobile',
  // ...
}